workspaces/src/result.rs

```diff
@@ -100,15 +105,31 @@ impl CallExecutionDetails {
         }
     }
 
-    /// Convert the execution details into a Result if its status is not a successful one.
-    /// Useful for checking if the call was successful and forwarding the error upwards.
-    fn try_into_result(self) -> Result<Self> {
+    /// Checks whether the transaction was executed successfully. This will return
+    /// a `Ok(())` if the transaction was successful, and a `Err(Error)` if it was not.
+    ///
+    /// This acts a convenience function to allow us to forward errors when we do not
+    /// care about the details of the transaction at all.
+    pub fn executed(&self) -> Result<()> {
+        self.try_into_success_value()?;
+        Ok(())
+    }
+
+    /// Checks whether the transaction was executed successfully. This will return
+    /// a `Ok(CallExecutionDetails)` if the transaction was successful, and a `Err(Error)`
+    /// if it was not.
+    pub fn into_result(self) -> Result<Self> {
         self.try_into_success_value()?;
         Ok(self)
     }
 
-    pub(crate) fn from_outcome(outcome: FinalExecutionOutcomeView) -> Result<Self> {
-        Self::from(outcome).try_into_result()
+    /// Checks whether the transaction execution failed. This will return
+    /// `Some((CallExeuctionDetails, Error))` if it did fail, and `None` if it succeeded.
+    pub fn err(self) -> Option<Error> {
```

Comment:
This comment is out of date. I would assume that this API would return something like `Option<CallExecutionDetails>` so perhaps this is an indication that the API is unintuitive. The fact that someone would have to check this error, match that the error kind is an execution error, then get details of the error (which is implicit from the previous step) seems like a huge leap for a developer.